src/reducer.js

```diff
@@ -26,8 +27,14 @@ export default function undoHistoryReducer(state=INITIAL_UNDO_HISTORY_STATE, act
     }
   case 'UNDO_HISTORY@ADD':
     {
+      const newUndoQueue = [omitBy(undoItem, isNil), ...undoQueue]
+
+      if (newUndoQueue.length === undoMemoryLimit + 1) {
+        newUndoQueue.pop() // Removing the first undoItem from the stuck
+      }
+
       return {
-        undoQueue: [omitBy(undoItem, isNil), ...undoQueue],
+        undoQueue: newUndoQueue,
         redoQueue: []
```

Comment:
Instead of everything here, just use:
undoQueue: [omitBy(undoItem, isNil), undoQueue.slice(0, undoQueueLimit-1)]